app/internal/wm/os_js.go

```diff
@@ -308,6 +308,25 @@ func (w *window) focus() {
 	w.requestFocus = true
 }
 
+func (w *window) keyboard(mode key.KeyboardMode) {
+	var m string
+	switch mode {
+	case key.KeyboardDefault:
+		m = "text"
+	case key.KeyboardNumeric:
+		m = "decimal"
+	case key.KeyboardEmail:
+		m = "email"
+	case key.KeyboardUrl:
+		m = "url"
+	case key.KeyboardTelephone:
+		m = "tel"
+	default:
+		m = "text"
+	}
+	w.tarea.Set("inputMode", m)
```

Comment:
`inputMode` isn't supported in some browsers (Firefox and old Safari). However, changing the `type` also affects the physical keyboard (not only the on-screen-keyboard).

It's possible to detect if `inputMode` isn't supported and use `type`. But, I think it's better to not change the `type` at all.